// draft-js-dnd-plugin/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _onDropFile = require('./modifiers/onDropFile');

var _onDropFile2 = _interopRequireDefault(_onDropFile);

var _onDropBlock = require('./modifiers/onDropBlock');

var _onDropBlock2 = _interopRequireDefault(_onDropBlock);

var _createDecorator = require('./createDecorator');

var _createDecorator2 = _interopRequireDefault(_createDecorator);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var store = {
  getReadOnly: undefined
};

var dndPlugin = function dndPlugin() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return {
    initialize: function initialize(_ref) {
      var getReadOnly = _ref.getReadOnly;

      store.getReadOnly = getReadOnly;
    },
    // Handle file drops
    handleDroppedFiles: (0, _onDropFile2.default)(config),
    // Handle any other drops (mostly blocks dragged and dropped across editor)
    handleDrop: (0, _onDropBlock2.default)(config),
    decorator: (0, _createDecorator2.default)({ store: store })
  };
};

exports.default = dndPlugin;
