// draft-js-video-plugin/lib/video/components/DefaultVideoComponent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var YOUTUBE_PREFIX = 'https://www.youtube.com/embed/';
var VIMEO_PREFIX = 'https://player.vimeo.com/video/';

var getSrc = function getSrc(_ref) {
  var srcID = _ref.srcID;
  var srcType = _ref.srcType;

  if (srcType === 'youtube') {
    return '' + YOUTUBE_PREFIX + srcID;
  }
  if (srcType === 'vimeo') {
    return '' + VIMEO_PREFIX + srcID;
  }
  return undefined;
};

var DefaultVideoCompoent = function DefaultVideoCompoent(props) {
  var blockProps = props.blockProps;

  var src = getSrc(blockProps);

  if (src) {
    return _react2.default.createElement(
      'div',
      { style: { width: '100%', height: '0', position: 'relative', paddingBottom: '56.25%' } },
      _react2.default.createElement('iframe', {
        style: { width: '100%', height: '100%', position: 'absolute', top: '0', left: '0' },
        src: src,
        frameBorder: '0',
        allowFullScreen: true
      })
    );
  }

  return _react2.default.createElement(
    'div',
    null,
    'invalid video source'
  );
};

DefaultVideoCompoent.propTypes = {
  blockProps: _react.PropTypes.object
};
exports.default = DefaultVideoCompoent;
